beacon_node/lighthouse_network/src/peer_manager/network_behaviour.rs

```diff
@@ -28,11 +28,49 @@ impl<TSpec: EthSpec> NetworkBehaviour for PeerManager<TSpec> {
         ConnectionHandler
     }
```

Comment:
I'm aware. We have three (or four?) custom behaviours and none is flagging this or the `addresses_of_peer` methods

Reply:
Yes it is not flagged unfortunately, see https://github.com/rust-lang/rust/issues/98990.

`new_handler` and `addresses_of_peer` will be removed in one of the next releases so migrating your behaviours now would be good. In case you haven't followed as closely, the four new callbacks should now allow you to very neatly implement connection management.

There are also two new behaviours on the way that already make use of this:

- https://github.com/libp2p/rust-libp2p/pull/3590
- https://github.com/libp2p/rust-libp2p/pull/3386